multiversx_sdk_core/transaction_factories/delegation_factory.py

```diff
@@ -0,0 +1,408 @@
+                           gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        parts = [
+            "changeServiceFee",
+            arg_to_string(service_fee)
+        ]
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=parts,
+            execution_gas_limit=self.config.delegation_ops + self.config.additional_gas_for_operations,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def modify_delegation_cap(self,
+                              sender: IAddress,
+                              delegation_contract: IAddress,
+                              delegation_cap: int,
+                              value: ITransactionValue,
+                              transaction_nonce: Optional[INonce] = None,
+                              gas_price: Optional[IGasPrice] = None,
+                              gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        parts = [
+            "modifyTotalDelegationCap",
+            arg_to_string(delegation_cap)
+        ]
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=parts,
+            execution_gas_limit=self.config.delegation_ops + self.config.additional_gas_for_operations,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def automatic_activation(self,
+                             sender: IAddress,
+                             delegation_contract: IAddress,
+                             set: bool,
+                             unset: bool,
+                             value: ITransactionValue,
+                             transaction_nonce: Optional[INonce] = None,
+                             gas_price: Optional[IGasPrice] = None,
+                             gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        parts = ["setAutomaticActivation"]
+
+        if set:
+            parts.append(arg_to_string('true'))
+
+        if unset:
+            parts.append(arg_to_string('false'))
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=parts,
+            execution_gas_limit=self.config.delegation_ops + self.config.additional_gas_for_operations,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def redelegate_cap(self,
+                       sender: IAddress,
+                       delegation_contract: IAddress,
+                       set: bool,
+                       unset: bool,
```

Comment:
same issue as explained for `automatic_activation`

Reply:
Same solution as above, we now have 2 methods called `set_redelegate_cap` and `unset_redelegate_cap`